internal/test/wsu/wsu_test.go

```diff
@@ -55,79 +53,255 @@ var (
+	//wmcbExeRegex searches for the path of the WMCB exe
+	wmcbExeRegex = regexp.MustCompile(`\/tmp\S*\/wmcb\.exe`)
+
+	// pinnedWMCBVM indicates which VM will use a pinned wmcb version
+	pinnedWMCBVM = 1
 )
 
-// createhostFile creates an ansible host file and returns the path of it
-func createHostFile(ip, password string) (string, error) {
+// getLatestReleasedArtifactURL returns the URL of the latest releases artifact containing the given name
+func getLatestReleasedArtifactURL(artifactName string) (string, error) {
+	resp, err := http.Get(wmcbReleasesURL)
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusOK {
+		return "", fmt.Errorf("recieved non OK status code: %v", resp)
+	}
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return "", err
+	}
+	// body is a JSON array and we need to get the browser_download_url, which is the last field in an object.
+	lines := strings.Split(string(body), "browser_download_url")
+	for i, line := range lines {
+		trimmedString := line[:strings.Index(line, "}")]
+		// trimmedString should look something like:
+		// ":"https://github.com/openshift/windows-machine-config-operator/releases/download/v0.3-alpha/wmcb.exe"
+		// we now need to get just the URL by splitting on \"
+		trimmedStringSplit := strings.Split(trimmedString, "\"")
+		if len(trimmedStringSplit) < 3 {
+			return "", fmt.Errorf("%v has unexpected length", trimmedStringSplit)
+		}
+		lines[i] = trimmedStringSplit[2]
+	}
+
+	for _, line := range lines {
+		// The latest release will be the closest to the beginning of the string
+		if strings.Contains(line, artifactName) {
+			return line, nil
+		}
+	}
+	return "", fmt.Errorf("could not find %s in release list", artifactName)
+}
+
+// createhostFile creates an ansible host file for the VMs we have spun up
+func createHostFile(vmList []e2ef.WindowsVM) (string, error) {
 	hostFile, err := ioutil.TempFile("", "testWSU")
 	if err != nil {
 		return "", fmt.Errorf("coud not make temporary file: %s", err)
 	}
 	defer hostFile.Close()
 
-	_, err = hostFile.WriteString(fmt.Sprintf(`[win]
-%s ansible_password='%s'
-[win:vars]
+	// Add each host to the host file
+	hostFileContents := "[win]\n"
+	for i := 0; i < len(vmList); i++ {
+		creds := vmList[i].GetCredentials()
+		hostFileContents += creds.GetIPAddress() + " " + "ansible_password='" + creds.GetPassword() + "'" + "\n"
+	}
+
+	// Add the common variables
+	hostFileContents += fmt.Sprintf(`[win:vars]
 ansible_user=Administrator
 cluster_address=%s
 ansible_port=5986
 ansible_connection=winrm
-ansible_winrm_server_cert_validation=ignore`, ip, password, clusterAddress))
+ansible_winrm_server_cert_validation=ignore
+`, clusterAddress)
+	_, err = hostFile.WriteString(hostFileContents)
 	return hostFile.Name(), err
 }
 
+// hashRemoteFile returns the hash of a file located at the given URL
+func hashRemoteFile(fileURL string) ([sha256.Size]byte, error) {
```

Comment:
Sorry, I should have caught this in the previous iteration.

What is the point in getting the hash in this manner? If there was a network error during the download, you will end up with an incorrect file and hash. You need to look at the github release and figure out the hash from there.

Reply:
This may be my lack of knowledge of network internals showing but how would a network error happen? I check for a 200 response and I believe packet loss shouldn't be possible over TCP.

I dont see a hash field in the github release api https://developer.github.com/v3/repos/releases/#list-releases-for-a-repository

Reply:
>This may be my lack of knowledge of network internals showing but how would a network error happen? I check for a 200 response and I believe packet loss shouldn't be possible over TCP.

It is just that you don't create your own source of truth on the fly but rather use the one that exists.

>I dont see a hash field in the github release api https://developer.github.com/v3/repos/releases/#list-releases-for-a-repository

You get it from the body of the github release similar to what @ravisantoshgudimetla is doing in his PR.